src/systems/particle_emitter/ParticleEmitter.cc

```diff
@@ -186,13 +224,52 @@ void ParticleEmitter::Configure(const Entity &_entity,
     components::ParticleEmitter(this->dataPtr->emitter));
 
   _ecm.CreateComponent(entity, components::Pose(pose));
+
+  // Advertise the topic to receive particle emitter commands.
+  const std::string kDefaultTopic =
+    "/model/" + this->dataPtr->model.Name(_ecm) + "/particle_emitter/" + name;
+  std::string topic = _sdf->Get<std::string>("topic", kDefaultTopic).first;
+  if (!this->dataPtr->node.Subscribe(
+         topic, &ParticleEmitterPrivate::OnCmd, this->dataPtr.get()))
+  {
+    ignerr << "Error subscribing to topic [" << topic << "]" << std::endl;
+    return;
+  }
+  igndbg << "Subscribed to " << topic << " for receiving particle emitter "
+         << "updates" << std::endl;
 }
 
 //////////////////////////////////////////////////
-void ParticleEmitter::PreUpdate(const ignition::gazebo::UpdateInfo &/*_info*/,
-    ignition::gazebo::EntityComponentManager &/*_ecm*/)
+void ParticleEmitter::PreUpdate(const ignition::gazebo::UpdateInfo &_info,
+    ignition::gazebo::EntityComponentManager &_ecm)
 {
   IGN_PROFILE("ParticleEmitter::PreUpdate");
+
+  if (!this->dataPtr->newDataReceived)
+    return;
+
+  // Nothing left to do if paused.
+  if (_info.paused)
+    return;
+
+  // Create component.
+  std::lock_guard<std::mutex> lock(this->dataPtr->mutex);
+
+  this->dataPtr->newDataReceived = false;
+
+  // Create an entity.
+  auto entity = _ecm.CreateEntity();
+  if (entity == kNullEntity)
+  {
+    ignerr << "Failed to create a particle emitter entity command" << std::endl;
+    return;
+  }
+
+  _ecm.CreateComponent(
+      entity,
+      components::ParticleEmitterCmd({this->dataPtr->userCmd}));
```

Comment:
As I mentioned in my other comment, I think it would be better to make the particle emitter command a component of the original emitter entity defined in `Configure` instead of creating a separate entity just for the command. I'm not sure what difficulties this change would introduce, though. I'm assuming that there may be difficulty deleting the component once the update has been applied and re-defining the component if new commands are given.

Reply:
done. a07c2ba39162191bacfe2fb74f1e1236124c0dac

I ran into an issue with removing the ParticleEmitterCmd component from the entity, which still needs to be resolved

Reply:
> I ran into an issue with removing the ParticleEmitterCmd component from the entity, which still needs to be resolved

Interesting, so the component is never removed? Maybe the new msgCmd component wasn't added to the entity yet, and we are trying to remove the component too soon? For example, would it make more sens to remove components after updating the emitters with the msgCmd in `RenderUtil::Update()`?:
https://github.com/ignitionrobotics/ign-gazebo/blob/a07c2ba39162191bacfe2fb74f1e1236124c0dac/src/rendering/RenderUtil.cc#L573-L577

Reply:
I think it's an issue with updating components on the gui side. Adding a OneTimeChange in the particle system when setting component data seem to made sure the new values are propagated to the gui. Updating the emitter properties should work now after f8a971d